source/javascripts/controllers/_MainController.js.erb

```diff
@@ -338,16 +349,48 @@ import _ from "underscore";
 				}
 			};
 
+			viewModel.openUpdateYmlInRepositoryModal = function() {
+				var variables = appService.getCurrentVariables();
+				appService.validateVariables(variables)
+					.then(function() {
+							viewModel.showUpdateYmlInRepositoryModal = true;
+					})
+			}
+
+			viewModel.closeUpdateYmlInRepositoryModal = function() {
+				$timeout(function() {
+					viewModel.saveProgress.reset();
+					viewModel.showUpdateYmlInRepositoryModal = false;
+				}, 0)
+			}
+
+			viewModel.onSaveToRepoComplete = function() {
+				$timeout(function() {
+					viewModel.closeUpdateYmlInRepositoryModal();
+				}, 2000)
+
+				$timeout(function() {
+					appService.afterAppConfigSaved();
```

Comment:
This should only be done in the Workflows/Env vars/Triggers tab. The bitrise.yml tab needs a separate method like this - see the success block of `appService.saveAppConfigYML`.